src/jl_uv.c

```diff
@@ -1085,6 +1085,25 @@ JL_DLLEXPORT int jl_queue_work(work_cb_t work_func, void *work_args, void *work_
     return 0;
 }
 
+JL_DLLEXPORT int jl_uv_update_timer_start(uv_loop_t* loop, uv_timer_t* handle,
+                                          uv_timer_cb cb, uint64_t timeout,
+                                          uint64_t repeat)
+{
+    JL_UV_LOCK();
+    int err = uv_timer_init(loop, handle);
+    if (err) {
+        // TODO: this codepath is currently not tested
+        free(handle);
```

Comment:
missing JL_UV_UNLOCK on this codepath, also probably better to keep the `free` call in Julia

Reply:
Hopefully my latest commit addresses both of these properly :smile: